components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -743,6 +1037,131 @@ private LocalClaim createLocalClaim(LocalClaimReqDTO localClaimReqDTO) {
         return new LocalClaim(localClaimReqDTO.getClaimURI(), attributeMappings, claimProperties);
     }
 
+    /**
+     * Import a claim dialect with related claims from an uploaded file.
+     *
+     * @param fileInputStream   InputStream representing the uploaded claim dialect file.
+     * @param fileDetail        Attachment object with metadata about the uploaded claim dialect file.
+     * @return a String representing the ID of the imported claim dialect.
+     */
+    public String importClaimDialectFromFile(InputStream fileInputStream, Attachment fileDetail) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(String.format("Importing claim dialect from file: %s", fileDetail.getDataHandler().getName()));
+        }
+
+        String dialectId;
+        try {
+            ClaimDialectConfiguration dialectConfiguration = getDialectFromFile(fileInputStream, fileDetail);
+            dialectId = dialectConfiguration.getId();
+
+            if (LOCAL_DIALECT_PATH.equals(dialectId)) {
+                throw handleClaimManagementClientError(
+                        Constant.ErrorMessage.ERROR_CODE_ERROR_IMPORTING_LOCAL_CLAIM_DIALECT, FORBIDDEN);
+            }
+
+            List<ExternalClaimReqDTO> externalClaimReqDTOList = dialectConfiguration.getExternalClaimReqDTOList();
+            String dialectURI = addClaimDialect(dialectConfiguration.getClaimDialectReqDTO());
+
+            importExternalClaims(dialectURI, externalClaimReqDTOList);
+
+            return dialectId;
+        } catch (ClaimMetadataException e) {
+            throw handleClaimManagementException(e, Constant.ErrorMessage.ERROR_CODE_ERROR_IMPORTING_CLAIM_DIALECT);
+        }
+    }
+    private void importExternalClaims(String dialectID, List<ExternalClaimReqDTO> externalClaimReqDTOList) {
+
+        List<ClaimErrorDTO> errors = new ArrayList<>();
+
+        for (ExternalClaimReqDTO externalClaimReqDTO : externalClaimReqDTOList) {
+            try {
+                addExternalClaim(dialectID, externalClaimReqDTO);
+            } catch (APIError e) {
+                ClaimErrorDTO claimErrorDTO = new ClaimErrorDTO(e.getResponseEntity());
+                claimErrorDTO.setClaimURI(externalClaimReqDTO.getClaimURI());
+                errors.add(claimErrorDTO);
+            }
+        }
+        if (!errors.isEmpty()) {
+            throw handleClaimManagementBulkClientError(Constant.ErrorMessage.ERROR_CODE_IMPORTING_EXTERNAL_CLAIMS,
+                    BAD_REQUEST, errors, String.valueOf(errors.size()), String.valueOf(externalClaimReqDTOList.size()));
+        }
+    }
+
+    private ClaimDialectConfiguration getDialectFromFile(InputStream fileInputStream, Attachment fileDetail)
+            throws ClaimMetadataException {
+
+        try {
+            FileContent claimFileContent = new FileContent(fileDetail.getDataHandler().getName(),
```

Comment:
Let's use `claimDialectFileContent` or `dialectFileContent`

Reply:
Changed to `claimDialectFileContent`